src/coreclr/src/jit/lsrabuild.cpp

```diff
@@ -3048,59 +3070,84 @@ void LinearScan::setDelayFree(RefPosition* use)
 
 //------------------------------------------------------------------------
 // BuildDelayFreeUses: Build Use RefPositions for an operand that might be contained,
-//                     and which need to be marked delayRegFree
+//                     and which may need to be marked delayRegFree
 //
 // Arguments:
-//    node      - The node of interest
+//    node       - The node of interest
+//    rmwNode    - The node that has RMW semantics (if applicable)
```

Comment:
I looked into this, and in doing so, I realized that I had made a set of changes to another branch that were somewhat independent of this branch, but which made it easier to duplicate the behavior of old and new (i.e. I was comparing these changes against those, rather than the actual unchanged sources). Those changes are in my https://github.com/CarolEidt/runtime/tree/LsraSmallMods branch, and these changes are included there as well. The crossgen diffs for frameworks and benchmarks for that branch relative to master are:

| Arch  | OS      | What                   | Delta            | Methods Improved | Methods Regressed |
| ----  | :--     | :--                    | --:              | --:              | --:               |
| Arm   | Windows | Crossgen fx+benchmarks | -1008 (-0.003%)  | 471              | 305               |
| Arm64 | Windows | Crossgen fx+benchmarks | -1276 (-0.002%)  | 251              | 283               |
| x64   | Windows | Crossgen fx+benchmarks | -1954 (-0.006%)  | 516              | 383               |
| x64   | Linux   | Crossgen fx+benchmarks | -1353 (-0.003%)  | 612              | 550               |
| x86   | Windows | Crossgen fx+benchmarks | 431 (0.002%)     | 232              | 253               |

So this change is not intrinsically related, but if you look at the old code in `registerIsAvailable` it had some special cases that were hard to duplicate in the new refactoring. With this change, we didn't require that special casing.